### PR TITLE
gapfiller: traverse subdirectories in patch

### PR DESCRIPTION
**Pull request by codeandkey**
The current gapfiller package patches some but not all (bwa) of the necessary perl sources.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gapfiller/package.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def url_for_version(self, version):
 
     def patch(self):
         with working_dir('.'):
-            files = glob.iglob("*.pl")
+            files = glob.glob("**.pl")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gapfiller/package.py]: Is there any reason to switch from iglob to glob?

Comment (codeandkey) [reply var/spack/repos/builtin/packages/gapfiller/package.py]: Oops; that was an extraneous change, however the performance won't change much as there are very few files it finds (less than 10 IIRC)

Comment (codeandkey) [reply var/spack/repos/builtin/packages/gapfiller/package.py]: I went ahead and reverted the function change.

File: var/spack/repos/builtin/packages/gapfiller/package.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def url_for_version(self, version):
 
     def patch(self):
         with working_dir('.'):
-            files = glob.iglob("*.pl")
+            files = glob.iglob("**.pl")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gapfiller/package.py]: First of all:
```suggestion
            files = glob.iglob("**.pl", recursive=True)
```
Also, it looks like support for this was added in Python 3.5, so this won't work for Python 2. You'll have to use `os.walk` instead. If you look in `lib/spack/llnl/util/filesystem.py`, I think you could use the `find` function here. It works pretty similarly to the UNIX command-line `find`.

Comment (codeandkey) [reply var/spack/repos/builtin/packages/gapfiller/package.py]: I modified the recipe to stick to the original `glob.glob` by explicitly adding the bwa subdirectory; as there are very few perl files it needs to find I think this is the most succinct approach.

**Pull request merged**